src/tools/kdb/command.h

```diff
@@ -0,0 +1,260 @@
+/**
+ * @file
+ *
+ * @brief Header for things used everywhere in the kdb tool
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#ifndef ELEKTRA_KDB_COMMAND_H
+#define ELEKTRA_KDB_COMMAND_H
+
+#include <colors.h>
+
+#include <kdb.h>
+#include <stdbool.h>
+#include <unistd.h>
+
+#define CLI_BASE_KEY "/sw/elektra/kdb/#0/current"
+
+/**
+ * @def ADD_BASIC_OPTIONS
+ * @brief This macro appends basic keys with option meta-data to a given key set.
+ * @param baseSpec KeySet to append to.
+ * @param baseKeyName The base key name for the appended keys.
+ */
+#define ADD_BASIC_OPTIONS(baseSpec, baseKeyName)                                                                                           \
+	ksAppendKey (baseSpec, keyNew (baseKeyName "/debug", KEY_META, "description",                                                      \
+				       "Give debug information or ask debug questions (in interactive mode).", KEY_META, "opt", "d",       \
+				       KEY_META, "opt/long", "debug", KEY_META, "opt/arg", "none", KEY_END));                              \
+	ksAppendKey (baseSpec, keyNew (baseKeyName "/verbose", KEY_META, "description", "Explain what is happening.", KEY_META, "opt",     \
+				       "v", KEY_META, "opt/long", "verbose", KEY_META, "opt/arg", "none", KEY_END));                       \
+	ksAppendKey (baseSpec, keyNew (baseKeyName "/version", KEY_META, "description", "Print version info.", KEY_META, "opt", "V",       \
+				       KEY_META, "opt/long", "version", KEY_META, "opt/arg", "none", KEY_END));                            \
+	ksAppendKey (baseSpec, keyNew (baseKeyName "/color", KEY_META, "description", "Print never/auto(default)/always colored output.",  \
+				       KEY_META, "opt", "C", KEY_META, "opt/arg/help", "WHEN", KEY_META, "opt/long", "color", KEY_META,    \
+				       "opt/arg", "required", KEY_END));                                                                   \
+	ksAppendKey (baseSpec,                                                                                                             \
+		     keyNew (baseKeyName "/nonewline", KEY_META, "description", "Suppress the newline at the end of the output.",          \
+			     KEY_META, "opt", "n", KEY_META, "opt/long", "no-newline", KEY_META, "opt/arg", "none", KEY_END));             \
+	ksAppendKey (spec, keyNew (COMMAND_SPEC_KEY (COMMAND_NAME) "/nullterm", KEY_META, "description", "Use binary 0 termination.",      \
+				   KEY_META, "opt", "0", KEY_META, "opt/long", "null", KEY_META, "opt/arg", "none", KEY_END));
+
+/**
+ * @def GET_OPT_KEY
+ * @brief This macro retrieves the key with a given name from an options key set. This is the more general version that GET_OPTION_KEY in
```

Comment:
```suggestion
 * @brief This macro retrieves the key with a given name from an options key set. This is the more general version that #GET_OPTION_KEY in
```

The `#` is needed so that doxygen generates a link to the other macro